src/sync/incomingSyncUtils.js

```diff
@@ -493,6 +497,16 @@ export const createOrUpdateRecord = (database, settings, recordType, record) =>
       numberSequence.addNumberToReuse(numberToReuse);
       break;
     }
+    case 'Report': {
+      internalRecord = {
+        id: record.ID,
+        title: record.title,
+        type: record.type,
+        _data: record.data,
+      };
+      database.update(recordType, internalRecord);
```

Comment:
yeah, I think you should do a check to make sure `data` is valid object that we want to accept and if not, reject the record (again, we can log to bugsnag later).

if we assign the value like this 
```
const someReport = database.objects("Report")[0]
someReport.data = {some: value}
```

Then the setter will be used.

If we set it with `update`, then the setter is not used.

IMO, we should keep both.. because being safe with your database is a good idea!

Reply:
Great, thanks for the explanation! Asking before commiting, this one we use checkIsObject  twice (before and after using the setter ).
```
case 'Report': {
      const reportData = database.objects('Report')[0];
      const setData = checkIsObject(record.data)
        ? reportData.data(record.data)
        : reportData.data({}); // Need to add a proper case solution for when record.data is not an Object
      internalRecord = {
        id: record.ID,
        title: record.title,
        type: record.type,
        _data: setData,
      };
      database.update(recordType, internalRecord);
      break;
  }
```

Reply:
Not quite. I think my example confused you, sorry. Hope the below will help!

`const reportData = database.objects('Report')[0];` - this will fetch the first `Report` record in the database. I was just using that as an example of accessing a realm object.

What you have is basically correct:

```
// Destructuring is nice!
const {ID:id, title, type, data} = record;

// We need to parse the data, so be safe and use a try
try{

// Parse the data to let us be able to check if it's correct and valid.
const parsedData = JSON.parse(data);

// Check if we should be setting this data. Even if it's parsed, it could still not be a JSONObject!
const shouldSetData = checkIsObject(JSON.parse(data))

// Now create the record we will put in the database. It's faster and easier to use .update() 
// here - but when we use that, the setter will NOT be used, and _data is a string field. If the
// parsed data is not a valid JSONObject, just set null for now.
const internalRecord = { id, title, type, _data: shouldSetData ? data : null }

// Finally, create the record
database.update(recordType, internalRecord)    
}catch(error){
    // Throw to parent, for now
    throw error
}
break;
```

Reply:
Thanks, much clearer btw :-) Done here 03ff6d7